app/controllers/admin/shifts_controller.rb

```diff
@@ -123,6 +123,38 @@ def create
     end
   end
 
+  def copy_to_next_week
+    end_of_week_param = DateTime.parse(params[:end_of_week])
+    start_of_week = end_of_week_param.beginning_of_week - 1.day
+    end_of_week = end_of_week_param.end_of_week - 1.day
+    new_shifts = []
+    Shift
+      .where(space_id: @space_id, start_datetime: start_of_week..end_of_week)
+      .each do |shift|
+        new_start = shift.start_datetime + 1.week
+        new_end = shift.end_datetime + 1.week
+        new_shift =
+          Shift.create(
+            space_id: @space_id,
+            start_datetime: new_start,
+            end_datetime: new_end,
+            reason: shift.reason,
+            pending: true
+          )
+        shift.users.each { |user| new_shift.users << user }
+        if !new_shift.save!
```

Comment:
I think it should be `!new_shift.save` otherwise it'll launch an error instead of redirecting (To be double checked)